src/System.Diagnostics.TraceSource/src/System/Diagnostics/CorrelationManager.cs

```diff
@@ -33,7 +34,13 @@ public Stack LogicalOperationStack
         {
             get
             {
-                return GetLogicalOperationStack();
+                ImmutableArray<object> elements = _idStack.Value.ToImmutableArray();
+                Stack newStack = new Stack();
+                for (int i = elements.Length-1; i >= 0; i--)
+                {
+                    newStack.Push(elements[i]);
+                }
+                return newStack;
```

Comment:
I'm not understanding how this works.  What if the developer does Trace.CorrelationManager.LogicalOperationStack.Push(12345)?  Doesn't that pushed value just get dropped on the floor?